chore(jans-core): avoid `jans-auth-model` dependency in `uma-rs-core` module

-------------------

### Description

We should remove `jans-auth-model` dependency from `uma-rs-core` module by moving few classes from `jans-auth-model` to jans-core. It is bad that jans-core depends on jans-auth.

```
[ERROR] Failed to execute goal on project uma-rs-core: Could not resolve dependencies for project io.jans:uma-rs-core:jar:1.0.4-SNAPSHOT: Failure to find io.jans:jans-auth-model:jar:1.0.4-SNAPSHOT in https://repo1.maven.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of mavencentral has elapsed or updates are forced -> [Help 1]
[ERROR] 
```
-------------------

### Prepare

- [ ] Read contribution guidelines
- [ ] Read license information

------------------

### Identified code changes


- [ ] task 1
- [ ] task 2
- [ ] task 3

-------------------

### Test cases and code coverage

- [ ] Write unit test to cover added/changed code
- [ ] Update integration tests to cover added/changed code

-------------------

### Document the changes

- [ ] task for updating user guides if needed
- [ ] task for updating installation and configuration guides if needed 
- [ ] task for updating developer documentation if needed
- [ ] task for updating technical documentation if needed

--- a/jans-core/uma-rs-core/pom.xml
+++ b/jans-core/uma-rs-core/pom.xml
@@ -97,16 +97,24 @@
             <artifactId>testng</artifactId>
         </dependency>
         <dependency>
-            <groupId>io.jans</groupId>
-            <artifactId>jans-auth-model</artifactId>
-        </dependency>
-        <dependency>
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
         </dependency>
         <dependency>
             <groupId>commons-codec</groupId>
             <artifactId>commons-codec</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-annotations</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-databind</artifactId>
         </dependency>
         <!-- Logging -->
         <dependency>

--- a/jans-core/uma-rs-core/src/main/java/io/jans/ca/rs/protect/Condition.java
+++ b/jans-core/uma-rs-core/src/main/java/io/jans/ca/rs/protect/Condition.java
@@ -5,6 +5,7 @@ import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.databind.JsonNode;
 
+import java.io.Serializable;
 import java.util.List;
 
 /**
@@ -12,16 +13,16 @@ import java.util.List;
  * @version 0.9, 18/04/2016
  */
 @JsonIgnoreProperties(ignoreUnknown = true)
-public class Condition {
+public class Condition implements Serializable {
 
     @JsonProperty(value = "httpMethods")
-    List<String> httpMethods;
+    private List<String> httpMethods;
     @JsonProperty(value = "scopes")
-    List<String> scopes;
+    private List<String> scopes;
     @JsonProperty(value = "scope_expression")
-    JsonNode scope_expression;
+    private transient JsonNode scopeExpression;
     @JsonProperty(value = "ticketScopes")
-    List<String> ticketScopes;
+    private List<String> ticketScopes;
 
     public List<String> getTicketScopes() {
         return ticketScopes;
@@ -48,11 +49,11 @@ public class Condition {
     }
 
     public JsonNode getScopeExpression() {
-        return scope_expression;
+        return scopeExpression;
     }
 
     public void setScopeExpression(JsonNode scopeExpression) {
-        this.scope_expression = scopeExpression;
+        this.scopeExpression = scopeExpression;
     }
 
     @Override
@@ -61,7 +62,7 @@ public class Condition {
         sb.append("Condition");
         sb.append("{httpMethods=").append(httpMethods);
         sb.append(", scopes=").append(scopes);
-        sb.append(", scope_expression=").append(scope_expression);
+        sb.append(", scopeExpression=").append(scopeExpression);
         sb.append(", ticketScopes=").append(ticketScopes);
         sb.append('}');
         return sb.toString();

--- a/jans-core/uma-rs-core/src/main/java/io/jans/ca/rs/protect/JsonLogicNode.java
+++ b/jans-core/uma-rs-core/src/main/java/io/jans/ca/rs/protect/JsonLogicNode.java
@@ -1,0 +1,67 @@
+package io.jans.ca.rs.protect;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author yuriyz
+ */
+@JsonPropertyOrder({"ticket"})
+public class JsonLogicNode {
+
+    private JsonNode rule;
+    private List<String> data;
+
+    public JsonLogicNode() {
+    }
+
+    public JsonLogicNode(JsonNode rule, List<String> data) {
+        this.rule = rule;
+        this.data = data;
+    }
+
+    @JsonIgnore
+    public boolean isValid() {
+        return data != null && !data.isEmpty() && rule != null;
+    }
+
+    @JsonProperty(value = "rule")
+    public JsonNode getRule() {
+        return rule;
+    }
+
+    public void setRule(JsonNode rule) {
+        this.rule = rule;
+    }
+
+    @JsonProperty(value = "data")
+    public List<String> getData() {
+        return data;
+    }
+
+    public void setData(List<String> data) {
+        this.data = data;
+    }
+
+    @JsonIgnore
+    public List<String> getDataCopy() {
+        if (data == null) {
+            return new ArrayList<>();
+        }
+        return new ArrayList<>(data);
+    }
+
+    @Override
+    public String toString() {
+        return "JsonLogicNode{" +
+                "rule=" + rule +
+                ", data=" + data +
+                '}';
+    }
+}
+

--- a/jans-core/uma-rs-core/src/main/java/io/jans/ca/rs/protect/RsResource.java
+++ b/jans-core/uma-rs-core/src/main/java/io/jans/ca/rs/protect/RsResource.java
@@ -4,8 +4,6 @@ import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.google.common.collect.Maps;
-import io.jans.as.model.uma.JsonLogicNode;
-import io.jans.as.model.uma.JsonLogicNodeParser;
 
 import java.io.Serializable;
 import java.util.List;
@@ -19,9 +17,9 @@ import java.util.Map;
 public class RsResource implements Serializable {
 
     @JsonProperty(value = "path")
-    String path;
+    private String path;
     @JsonProperty(value = "conditions")
-    List<Condition> conditions;
+    private List<Condition> conditions;
     @JsonProperty(value = "iat")
     private Integer iat;
     @JsonProperty(value = "exp")
@@ -72,13 +70,21 @@ public class RsResource implements Serializable {
     public List<String> getScopesForTicket(String httpMethod) {
         Condition condition = getConditionMap().get(httpMethod);
         if (condition.getScopeExpression() != null) {
-            final JsonLogicNode node = JsonLogicNodeParser.parseNode(condition.getScopeExpression().toString());
+            final JsonLogicNode node = parseNode(condition.getScopeExpression().toString());
             if (node != null) {
                 return node.getData(); // return all scopes defined in "data" of json object
             }
         }
         return condition.getTicketScopes() != null && !condition.getTicketScopes().isEmpty() ?
                 condition.getTicketScopes() : condition.getScopes();
+    }
+
+    public static JsonLogicNode parseNode(String json) {
+        try {
+            return Jackson.createJsonMapper().readValue(json, JsonLogicNode.class);
+        } catch (Exception e) {
+            return null;
+        }
     }
 
     public Map<String, Condition> getConditionMap() {

